exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -514,65 +678,190 @@ func (e *Exchange) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Sub
 	if err := s.Validate(e.GetTradingRequirements()); err != nil {
 		return nil, err
 	}
-
-	fPair, err := e.FormatExchangeCurrency(s.Pair, s.AssetType)
+	var err error
+	s.Pair, err = e.FormatExchangeCurrency(s.Pair, s.AssetType)
+	if err != nil {
+		return nil, err
+	}
+	oTypeString, err := OrderTypeString(s.Type)
 	if err != nil {
 		return nil, err
 	}
-	response, err := e.PlaceOrder(ctx,
-		fPair.String(),
-		s.Price,
-		s.Amount,
-		false,
-		s.Type == order.Market,
-		s.Side.IsLong())
+	tif, err := TimeInForceString(s.TimeInForce)
 	if err != nil {
 		return nil, err
 	}
-	return s.DeriveSubmitResponse(strconv.FormatInt(response.OrderNumber, 10))
+	switch s.AssetType {
+	case asset.Spot:
+		var smartOrder bool
+		switch s.Type {
+		case order.Stop, order.StopLimit, order.TrailingStop:
+			smartOrder = true
+		case order.Limit, order.Market, order.LimitMaker, order.UnknownType:
+		default:
+			return nil, fmt.Errorf("%v order type %v is not supported", order.ErrTypeIsInvalid, s.Type)
+		}
+		if smartOrder {
+			sOrder, err := e.CreateSmartOrder(ctx, &SmartOrderRequestRequest{
+				Symbol:        s.Pair,
+				Type:          oTypeString,
+				Side:          s.Side.String(),
+				AccountType:   accountTypeString(s.AssetType),
+				Price:         s.Price,
+				StopPrice:     s.TriggerPrice,
+				Quantity:      s.Amount,
+				ClientOrderID: s.ClientOrderID,
+				TimeInForce:   tif,
```

Comment:
🚧 🤷‍♂️ Clarification please:
* It seems we're consistently using strings in our request structs here, and constantly doing the same conversions in many request methods.
Can we make local types for `accountType`, `orderType`, and `timeInForce`?
* And can side not be an order.Side natively if we're just going to string it?
* For accountType, this'd mean moving accountType String to a method on `accountType`, which seems sane in-line with the others.
* 🌴 Obviously this change applies throughout this PR/file.
```suggestion
				Type:          orderType(s.Type),
				Side:          s.Side,
				AccountType:   accountType(s.AssetType),
				Price:         s.Price,
				StopPrice:     s.TriggerPrice,
				Quantity:      s.Amount,
				ClientOrderID: s.ClientOrderID,
				TimeInForce:   timeInForce(s.TimeInForce),
```

Reply:
I have made these functions un-exported but left the naming as it is because, I think, it is more explanatory than what you suggested.
Thank you!